rpc/src/rpc.rs

```diff
@@ -3436,12 +3437,7 @@ pub mod rpc_full {
                     Error::invalid_request()
                 })?
                 .iter()
-                .map(|(slot, sample)| RpcPerfSample {
-                    slot: *slot,
-                    num_transactions: sample.num_transactions,
-                    num_slots: sample.num_slots,
-                    sample_period_secs: sample.sample_period_secs,
-                })
+                .map(|(slot, sample)| rpc_perf_sample_from_perf_sample(*slot, sample))
```

Comment:
Done.

By the way, `Blockstore::get_recent_perf_samples()` is currently defined as
```rust
    pub fn get_recent_perf_samples(&self, num: usize) -> Result<Vec<(Slot, PerfSample)>> {
```
But it can easily return an iterator instead:
```rust
    pub fn get_recent_perf_samples(&self, num: usize) -> Result<impl Iterator<Item = (Slot, PerfSample)> + '_> {
```

This would remove the packing/unpacking into/from a vector.
Caller can always do a `.collect()` call.
I'll send a PR to fix this.